nested_lookup/lookup_api.py

```diff
@@ -29,28 +31,134 @@ def _nested_delete(document, key):
     return document
 
 
-def nested_update(document, key, value, in_place=False):
+def nested_update(document:dict, key:str, value:object, in_place:bool=False, treat_list_as_element:bool = False) -> dict:
```

Comment:
Please change the argument name to something like `treat_as_element` since the value can be any type (list or dict). i suggest you to have the default value as True.

Reply:
Changed the name and made the default value Trueto have a better backwards portability of the solution.